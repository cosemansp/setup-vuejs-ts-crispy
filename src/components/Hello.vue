<template>
  <div class="hello">
    <h1>{{ msg }}</h1>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import Component from 'vue-class-component'
// more info: https://github.com/vuejs/vue-class-component

@Component
export default class App extends Vue {
  msg = 'Welcome to Your Vue.js App'
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h1, h2 {
  font-weight: normal;
}
</style>
