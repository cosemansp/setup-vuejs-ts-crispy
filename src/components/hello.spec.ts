import { mount, mountConfig } from '../../test/helper';

// Need to import with require because there is no typings
// available via *.vue files
// const Hello = require('./Hello.vue');

import Hello from './hello';
import HelloWithTemplate from './hello.vue';

describe('Hello Component', () => {

  it('should have a default "msg" set to "Welcome"', () => {
    const component = new Hello();
    expect(component.msg).toMatch('Welcome');
  });

  it('should renders correctly', () => {
    // render component as a single unit
    const fixture = mount(HelloWithTemplate);
    fixture.setData({ msg: 'Hello World' });
    expect(fixture).toMatchSnapshot();
  });

  it('should renders correctly with childs', () => {
    // render component with children
    const fixture = mount(HelloWithTemplate);
    expect(fixture).toMatchSnapshot();
  });

  it('should renders config', () => {
    const componentConfig = {
      template: `
        <div>
          <hello ref="component"></hello>
        </div>
      `,
      components: {
        hello: HelloWithTemplate,
      },
    };
    const fixture = mountConfig(componentConfig);
    expect(fixture).toMatchSnapshot();
    expect(fixture.vm.$refs.component).toBeTruthy();
    expect(fixture.vm.$refs.component.msg).toMatch('Welcome');
  });
});

