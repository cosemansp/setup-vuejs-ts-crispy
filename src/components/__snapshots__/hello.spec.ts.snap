// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello Component should renders config 1`] = `
<div>
  <div class="hello">
    <h1>Welcome to Your Vue.js App</h1>
  </div>
</div>
`;

exports[`Hello Component should renders correctly 1`] = `
<div class="hello">
  <h1>Hello World</h1>
</div>
`;

exports[`Hello Component should renders correctly with childs 1`] = `
<div class="hello">
  <h1>Welcome to Your Vue.js App</h1>
</div>
`;
